src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,379 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7798.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE77B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE77F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7838.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7878.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE78D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7954.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE79A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7AF8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7BF8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7C64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7C94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7CE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7D40.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7D94.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7FD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/func_80AE7FDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Rl/EnRl_Draw.s")
+void func_80AE7798(EnRl* this, GlobalContext* globalCtx);
+void func_80AE77B8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE77F8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7838(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7C64(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7C94(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7CE8(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7D40(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7FD0(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7FDC(EnRl* this, GlobalContext* globalCtx);
+void func_80AE7D94(EnRl* this, GlobalContext* globalCtx);
+
+UNK_PTR D_80AE81A0[] = { 0x06003620, 0x06003960, 0x06003B60 };
+
+extern SkeletonHeader D_06007B38;
+extern AnimationHeader D_06000A3C;
+extern AnimationHeader D_06000830;
+extern AnimationHeader D_0600040C;
+
+void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnRl* this = THIS;
+
+    SkelAnime_Free(&this->skelAnime, globalCtx);
+}
+
+void func_80AE72D0(EnRl* this) {
+    s32 pad[3];
+    s16* unk_192 = &this->unk_192;
+    s16* unk_190 = &this->unk_190;
+
+    if (DECR(*unk_192) == 0) {
+        *unk_192 = Math_Rand_S16Offset(60, 60);
+    }
+
+    *unk_190 = *unk_192;
+    if (*unk_190 > 2) {
+        *unk_190 = 0;
+    }
+}
```

Comment:
Based on my comment on `D_80AE81A0` above and `unk_190` in `func_80AE7D94` below I think it's safe to say `unk_192` is a timer for eye blinking.
when that timer is > 2 the chosen texture index is 0 aka "open", when the timer is 1 or 2 the texture index is 1 or 2 aka in order half open/closed